### PR TITLE
docs: Update create compound component docs

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Updates create compound component docs to use new model utility functions.

## Release Category
Documentation

---

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)
- [x] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [x] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

Use the rich diff feature to better understand the changes.

### CODE REVIEW

File: modules/docs/mdx/CREATING_COMPOUND_COMPONENTS.mdx [hunk 1]
```diff
@@ -728,73 +681,167 @@ export const useExpandableControls = ({state}: DisclosureModel, elemProps: {}) =
...
+
+export const useExpandableControls = createElemPropsHook(useDisclosureModel)(({state}) => {
+  return {
+    'aria-controls': state.id,
+    'aria-expanded': state.visible,
+  };
+});
+```
+
+The `createElemPropsHook` takes the model hook and a elem props hook body as arguments. The hook
```
Comment (jamesfan) [inline modules/docs/mdx/CREATING_COMPOUND_COMPONENTS.mdx:700]:
```suggestion
`createElemPropsHook` takes the model hook and an elem props hook body as arguments. The hook
```

File: modules/docs/mdx/CREATING_COMPOUND_COMPONENTS.mdx [hunk 2]
```diff
@@ -728,73 +681,167 @@ export const useExpandableControls = ({state}: DisclosureModel, elemProps: {}) =
...
 );
+```
 
-return (
-  <Element ref={ref} {...props}>
-    {children}
-  </Element>
+Notice we still need to use `mergeProps` to compose the behavior of our two elemProps hooks? A
+`composeHooks` function was created to handle this common composition use-case. `composeHooks` takes
+two or more elemProps hooks and returns a new hook with all props merged for us:
```
Comment (jamesfan) [inline modules/docs/mdx/CREATING_COMPOUND_COMPONENTS.mdx:740]:
```suggestion
Notice we still need to use `mergeProps` to compose the behavior of our two elemProps hooks?
`composeHooks` was created to handle this common composition use case. `composeHooks` takes two or
more elemProps hooks and returns a new hook with all props merged for us:
```

**Review by jamesfan — COMMENTED** [review]
I focused more on reviewing the content for issues rather than the code snippets for syntax and logical errors (I didn't see anything in what I skimmed).

A couple things:

1. I assume the base branch of this PR should be updated to use `master` instead of `prerelease/major`?
2. It would be nice if the Codesandbox links in this document could be replaced with actual examples within our documentation (we have precedence now for including examples outside of component documentation in our [Style Props docs](https://github.com/Workday/canvas-kit/blame/master/modules/docs/mdx/style-props/STYLE_PROPS.mdx#L54)), though that could be saved for a follow-up PR.

**Comment by NicholasBoll** [comment]
> @NicholasBoll I focused more on reviewing the content for issues rather than the code snippets for syntax and logical errors (I didn't see anything in what I skimmed).
> 
> A couple things:
> 
> 1. I assume the base branch of this PR should be updated to use `master` instead of `prerelease/major`?
> 2. It would be nice if the Codesandbox links in this document could be replaced with actual examples within our documentation (we have precedence now for including examples outside of component documentation in our [Style Props docs](https://github.com/Workday/canvas-kit/blame/master/modules/docs/mdx/style-props/STYLE_PROPS.mdx#L54)), though that could be saved for a follow-up PR.

1. I updated the target branch. It was targeting v8 before we merged.
2. Our examples don't handle multiple files. The code could be updated to all be in one file, but it would be very long. I think the most interesting part of the examples is the code and not the working UI. I think codesandbox is the best way to illustrate everything working together.

**Review by mannycarrera4 — APPROVED** [review]

**Pull request merged**